invokeai/app/services/shared/invocation_context.py

```diff
@@ -434,6 +437,55 @@ def search_by_attrs(
             model_format=format,
         )
 
+    def download_and_cache_model(
+        self,
+        source: str | AnyHttpUrl,
+    ) -> Path:
+        """
+        Download the model file located at source to the models cache and return its Path.
+
+        This can be used to single-file install models and other resources of arbitrary types
+        which should not get registered with the database. If the model is already
+        installed, the cached path will be returned. Otherwise it will be downloaded.
+
+        Args:
+            source: A model path, URL or repo_id.
+
+        Returns:
+            Path to the downloaded model
+        """
+
+        return self._services.model_manager.install.download_and_cache_model(source=source)
```

Comment:
There is a type mismatch between these source args: `str` vs. `str  | AnyHttpUrl`

Reply:
Fixed.